### PR TITLE
Support Markdown in docs (#376)

### LINKED ISSUE #376: Add support for Markdown file rendering in Sphinx build docs
The two goals of this are:
- Allow documentation to be written in markdown
- Allow including the top level README.md into the rendered output

A proof of concept exists in hwine/sphinx-markdown, with rendered docs at https://hwine.github.io/sphinx-markdown/

The initial landing (#374 & #375) broke document publishing, so was reverted by 3c2e04b56166d7573987f5de7d989d69e9feb50f.

New plan:
- [x] initially land all work in a [fork](https://github.com/hwine/frost)
- [x] verify publishing works from fresh clone
- [x] verify publishing changes works by viewing rendered content at https://hwine.github.io/frost
- [x] create PR for the set of changes

### PR DESCRIPTION
**Pull request by hwine**
Markdown files now supported during Sphinx processing. Top level `README.md` is now included as "Overview" of rendered docs.

You can view the rendered and published-by-travis docs in my fork: <https://hwine.github.io/frost/>

A few tweaks were needed to get `README.md` to render well via Sphinx. There are some minor differences in the syntax highlighting of code blocks remaining.

Two "syntactic sugar" changes:
- new make target `doc-preview` that serves the locally produced sphinx rendered docs
- new "badge" for the `README.md` file that points to the rendered docs.

Fixes #376

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,6 +42,9 @@ doc-build: check_venv
 	type sphinx-build || { echo "please run `make install-docs` to build docs"; false; }
 	make -C docs html
 
+doc-preview: check_venv doc-build
+	python3 -m http.server --directory docs/_build/html/
+
```
Comment (smarnach) [inline Makefile:47]: We could also consider https://pypi.org/project/sphinx-autobuild/. (I've never used it, and just learned about its existence, so maybe that's for a separate PR.)

**Review by smarnach — APPROVED** [review]
Looking good to me.

I've taken the freedom to fix some links in the README file on this branch, since I noticed they were broken in [the rendered documentation on your fork](https://hwine.github.io/frost/readme-include.html#design). It turned out that these links have been broken all along, so it's actually an unrelated change, but we may just as well fix it as part of this PR.

**Review by g-k — APPROVED** [review]
r+

+1 on using sphinx-autobuild (here or in a separate PR)

File: Makefile (hunk 1 above)
Comment (hwine) [reply Makefile:47]: > We could also consider https://pypi.org/project/sphinx-autobuild/.

Oooo, nice find! TIL -- if no one else does, I'll try to integrate that this afternoon PT (I'm only doing 1/2 day today). If it gets messy, I'll land this as-is.

> I've taken the freedom to fix some links in the README file on this branch,

Much appreciated! IMO, PRs are like a wiki -- no permission to modify needed!

**Pull request merged**